packages/platform-sdk-nano/src/services/transaction.test.ts

```diff
@@ -0,0 +1,44 @@
+import "jest-extended";
+
+import { Signatories } from "@arkecosystem/platform-sdk";
+import nock from "nock";
+
+import { identity } from "../../test/fixtures/identity";
+import { createConfig } from "../../test/helpers";
+import { SignedTransactionData } from "../dto/signed-transaction";
+import { TransactionService } from "./transaction";
+
+let subject: TransactionService;
+
+beforeEach(async () => (subject = await TransactionService.__construct(createConfig())));
+
+afterEach(() => nock.cleanAll());
+
+beforeAll(() => nock.disableNetConnect());
+
+describe("TransactionService", () => {
+	describe("#transfer", () => {
+		it("should sign transaction", async () => {
+			nock("https://proxy.nanos.cc/")
+				.post("/proxy/")
+				.reply(200, require(`${__dirname}/../../test/fixtures/client/account-info.json`));
+
+			const result = await subject.transfer({
+				signatory: new Signatories.Signatory(
+					new Signatories.MnemonicSignatory({
+						signingKey: identity.mnemonic,
```

Comment:
@faustbrian Can object creation be simplified? Maybe some of the arguments interfaces

Reply:
All of this is handled by the `SignatoryService` outside of tests.

Reply:
Can MnemonicSignatory inherit from Signatory to avoid double-constructor?

Reply:
No, it's intentional like this because the implementations for each signatory differ but we don't want to hint 10 different interfaces/classes. It also provides some protection against calling wrong methods and helpers to determine what kind of signing method should be used without having to litter code with `signatory instanceof XXX` assertions.

None of this is of any concern outside of tests because the `SignatoryService` handles this construction, it's not something that will ever be constructed manually outside of tests.